libpod/options.go

```diff
@@ -979,6 +979,25 @@ func WithStaticIP(ip net.IP) CtrCreateOption {
 	}
 }
 
+// WithLogDriver sets the log driver for the container
+func WithLogDriver(driver string) CtrCreateOption {
+	return func(ctr *Container) error {
+		if ctr.valid {
+			return ErrCtrFinalized
+		}
+		if driver == "" {
+			return errors.Wrapf(ErrInvalidArg, "log driver must be set")
+		}
+		if driver != "journald" && driver != "k8s-file" && driver != "json-file" {
```

Comment:
Can we use the constants here, instead of comparing against strings?
Also, probably best to make it a switch